Add vegetation standalone documentation pages (#1437)
CliMA/ClimaLand.jl: docs/src/standalone/pages/vegetation/soil_moisture_stress/piecewise_model.md

# Piecewise Soil Moisture Stress

This page documents the **piecewise** (linear/threshold with curvature) soil moisture stress function used in ClimaLand's vegetation module. The stress factor $\beta \in [0,1]$ scales leaf photosynthesis (and thus stomatal conductance) as soil moisture declines, following a simple, interpretable formulation. See [Egea2011](@citet), the ClimaLand code is [here](https://github.com/CliMA/ClimaLand.jl/blob/main/src/standalone/Vegetation/soil_moisture_stress.jl).

---

## Summary
- **Purpose:** convert soil wetness to a *stress multiplier* that reduces assimilation and stomatal conductance.
- **Inputs:** volumetric soil water content $\theta$ (or relative water content), model thresholds.
- **Outputs:** scalar stress factor $\beta$ applied within the photosynthesis/stomatal sub-models.

---

## Model formulation

> Let $\theta$ be the volumetric water content of the plant-available root zone (or a weighted layer average if roots span multiple layers). Define two thresholds:

Reviewer: ```suggestion
Let $\theta$ be the volumetric water content at some depth. Define two thresholds:
```
